src/posix/platform/openthread-posix-config.h

```diff
@@ -101,4 +101,28 @@
 #define OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE 0
 #endif
 
+#ifdef __APPLE__
+
+/**
+ * Use built-in utun driver on mac OS
+ */
+#define OPENTHREAD_POSIX_CONFIG_MACOS_UTUN 1
+
+/**
+ * Use open-source tun driver on mac OS
+ */
+#define OPENTHREAD_POSIX_CONFIG_MACOS_TUN 2
+
+/**
+ * @def OPENTHREAD_POSIX_MACOS_TUN_OPTION
+ *
+ * This setting configures which tunnel driver to use.
+ *
+ */
+#ifndef OPENTHREAD_POSIX_MACOS_TUN_OPTION
+#define OPENTHREAD_POSIX_MACOS_TUN_OPTION OPENTHREAD_POSIX_CONFIG_MACOS_UTUN
```

Comment:
```suggestion
#define OPENTHREAD_POSIX_CONFIG_MACOS_TUN_OPTION OPENTHREAD_POSIX_CONFIG_MACOS_UTUN
```
nit: to be consistent with other config options in this file?

Reply:
Should OPENTHREAD_POSIX_CONFIG_MACOS_UTUN be renamed to `OT_*`?

Reply:
Made suggested changes.  Didn't see (didn't actually look for) guidance for why the options should be OT_POSIX_* instead of OPENTHREAD_POSIX_*, but I see that's being done for the RCP bus configuration, so I'm happy to be consistent with that.